tlslite/utils/python_tripledes.py

```diff
@@ -0,0 +1,491 @@
+            self._r = [x ^ y for x, y in zip(self._r, self._kn[iteration])]
+            b = [self._r[:6], self._r[6:12], self._r[12:18], self._r[18:24],
+                 self._r[24:30], self._r[30:36], self._r[36:42], self._r[42:]]
+
+            # Permutate b[1] to b[8] using the S-Boxes
+            bn = [0] * 32
+            pos = 0
+            for j in range(8):
+                # Work out the offsets
+                m = (b[j][0] << 1) + b[j][5]
+                n = (b[j][1] << 3) + (b[j][2] << 2) + (b[j][3] << 1) + b[j][4]
+
+                # Find the permutation value
+                v = Des.__sbox[j][(m << 4) + n]
+
+                # Turn value into bits, add it to result: bn
+                bn[pos] = (v & 8) >> 3
+                bn[pos + 1] = (v & 4) >> 2
+                bn[pos + 2] = (v & 2) >> 1
+                bn[pos + 3] = v & 1
+
+                pos += 4
+
+            # Permutate the concatination of b[1] to b[8] (bn)
+            self._r = self.__permutate(Des.__p, bn)
+
+            # Xor with _l[i - 1]
+            self._r = [x ^ y for x, y in zip(self._r, self._l)]
+            self._l = temp_r
+
+            iteration += iteration_adjustment
+
+        # Final permutation of _r[16]_l[16]
+        self._final = self.__permutate(Des.__fp, self._r + self._l)
+        return self._final
+
+    def crypt(self, data, crypt_type):
+        """Crypt the data in blocks, running it through des_crypt()."""
+
+        iv = self.__string_to_bitlist(self.iv)
+
+        # Split the data into blocks, crypting each one seperately
+        i = 0
+        result = []
+        while i < len(data):
+            # Test code for caching encryption results
+            block = self.__string_to_bitlist(data[i:i+8])
+
+            # Xor with iv if using CBC mode
+            if crypt_type == Des.ENCRYPT:
+                block = [x ^ y for x, y in zip(block, iv)]
+
+            processed_block = self.__des_crypt(block, crypt_type)
+
+            if crypt_type == Des.DECRYPT:
+                processed_block = [x ^ y for x, y in zip(processed_block, iv)]
+                iv = block
+            else:
+                iv = processed_block
+
+            # Add the resulting crypted block to our list
+            result.append(self.__bitlist_to_string(processed_block))
+            i += 8
+
+        # Return the full crypted string
+        return b''.join(result)
+
+#############################################
+#               Triple DES                  #
+#############################################
+
+
+class Python_TripleDES(_baseDes):
+    """Triple DES encryption/decrytpion class.
+
+    This algorithm uses the DES-EDE3 (when a 24 byte key is supplied) or
+    the DES-EDE2 (when a 16 byte key is supplied) encryption methods.
+    Supports CBC (Cypher Block Chaining) mode.
+    """
+    def __init__(self, key, iv=None):
+        super(Python_TripleDES, self).__init__(iv)
```

Comment:
so the problem with this is that if we fix the `stacklevel` to 3 in warn, it will report this line as the problematic one

we can fix it in two ways, either change the `_guard_against_unicode()` to detect this case, or call `_guard_against_unicode()` on `iv` from here, before calling `super()`, the latter is likely easier to do